mmdet/models/dense_heads/corner_head.py

```diff
@@ -0,0 +1,714 @@
+                      with_guiding_shift=False,
+                      with_centripetal_shift=False):
+        b, _, h, w = feat_shape
+        img_h, img_w = img_shape[:2]
+
+        width_ratio = float(w / img_w)
+        height_ratio = float(h / img_h)
+
+        gt_tl_heatmap = torch.zeros([b, self.num_classes, h, w],
+                                    dtype=feat_dtype,
+                                    device=feat_device)
+        gt_br_heatmap = torch.zeros([b, self.num_classes, h, w],
+                                    dtype=feat_dtype,
+                                    device=feat_device)
+        gt_tl_offset = torch.zeros([b, 2, h, w],
+                                   dtype=feat_dtype,
+                                   device=feat_device)
+        gt_br_offset = torch.zeros([b, 2, h, w],
+                                   dtype=feat_dtype,
+                                   device=feat_device)
+
+        if with_corner_emb:
+            match = []
+
+        if with_guiding_shift:
+            gt_tl_guiding_shift = torch.zeros([b, 2, h, w],
+                                              dtype=feat_dtype,
+                                              device=feat_device)
+            gt_br_guiding_shift = torch.zeros([b, 2, h, w],
+                                              dtype=feat_dtype,
+                                              device=feat_device)
+        if with_centripetal_shift:
+            gt_tl_centripetal_shift = torch.zeros([b, 2, h, w],
+                                                  dtype=feat_dtype,
+                                                  device=feat_device)
+            gt_br_centripetal_shift = torch.zeros([b, 2, h, w],
+                                                  dtype=feat_dtype,
+                                                  device=feat_device)
+
+        for b_id in range(b):
+            corner_match = []
+            for box_id in range(len(gt_labels[b_id])):
+                tl_x, tl_y, br_x, br_y = gt_bboxes[b_id][box_id]
+                ct_x = (tl_x + br_x) / 2.0
+                ct_y = (tl_y + br_y) / 2.0
+                label = gt_labels[b_id][box_id]
+
+                ftlx = float(tl_x * width_ratio)
+                fbrx = float(br_x * width_ratio)
+                ftly = float(tl_y * height_ratio)
+                fbry = float(br_y * height_ratio)
+                fctx = float(ct_x * width_ratio)
+                fcty = float(ct_y * height_ratio)
+
+                tl_x_idx = int(min(ftlx, w - 1))
+                br_x_idx = int(min(fbrx, w - 1))
+                tl_y_idx = int(min(ftly, h - 1))
+                br_y_idx = int(min(fbry, h - 1))
+
+                width = ceil(fbrx - ftlx)
+                height = ceil(fbry - ftly)
+
+                radius = gaussian_radius((height, width), min_overlap=0.3)
+                radius = max(0, int(radius))
+
+                gen_gaussian_target(gt_tl_heatmap[b_id, label],
```

Comment:
Inplace modifying tensor value is not recommended.
I suggest using an out of place version of this function.